src/web.cpp

```diff
@@ -1077,7 +1077,8 @@ void handle_firmware_upload()
         {
             uint32_t maxSketchSpace = (ESP.getFreeSketchSpace() - 0x1000) & 0xFFFFF000;
             RINFO("Available space for upload: %lu", maxSketchSpace);
-            if (!Update.begin(maxSketchSpace, U_FLASH))
+            RINFO("Flash chip speed %d MHz", ESP.getFlashChipSpeed() / 1000000);
+            if (!Update.begin((firmwareSize > 0) ? firmwareSize : maxSketchSpace, U_FLASH))
```

Comment:
What made you decide to make this change to the if statement?

Reply:
When originally implemented I was not sending firmware size or MD5 up to the server, so the `Update.begin()` was being passed the maxSketchSpace.  The Updater code then comes to an "unexpected end" when the data runs out before it has reached the "size" expected.  It works fine, but as we now know the firmware size, the correct thing to do is pass that size into `Update.begin()`.  I have the conditional just in case there is a scenario where we don't know it (there is no firm requirement that the MD5 / firmware size is send before the file upload).